rust/src/storage/s3/mod.rs

```diff
@@ -22,10 +23,107 @@ use uuid::Uuid;
 
 pub mod dynamodb_lock;
 
-const AWS_S3_ASSUME_ROLE_ARN: &str = "AWS_S3_ASSUME_ROLE_ARN";
-const AWS_S3_ROLE_SESSION_NAME: &str = "AWS_S3_ROLE_SESSION_NAME";
+const AWS_ENDPOINT_URL: &str = "AWS_ENDPOINT_URL";
 const AWS_WEB_IDENTITY_TOKEN_FILE: &str = "AWS_WEB_IDENTITY_TOKEN_FILE";
```

Comment:
Why are these also not considered options for use with the S3StorageOptions? :confused:  (Particularly endpoint url since that's got some behavioral meaning later on)

Reply:
i actually started w endpoint url as an option but that var is only set when testing against localstack and only used to create the region. given that, i felt like it was awkward to include that as part of the public api. we already set the env var in ci so i feel like omitting it from the options struct is a cleaner public api for consumers. i dont think anybody consuming delta-rs should need to use it

Reply:
RE: `AWS_ENDPOINT_URL` - I moved this to S3StorageOptions in the latest commit in case there are other use cases for custom S3 endpoints. The URL is only used for creating the `region`, but I added an unused `_endpoint_url` field for information/debug needs.

RE: `AWS_WEB_IDENTITY_TOKEN_FILE` - these must be set in the env for the rusoto web identity provider to work the way it is called today (`from_k8s_env`). I went ahead added these to the `s3_storage_options` keys along w/ some code to set them in env if they are passed in options map.

Reply:
`AWS_ENDPOINT_URL` will be required for anybody working with an S3 compatible API but not using S3 directly, which is why I think it should be included. Yes we're using it for localstack, but excluding it from the options limits the potential for users to use non-S3 storage providers with this code :shrug:

Reply:
Cool. Its in the options now so I think we are good.